modules/internal/ChapelUtil.chpl

```diff
@@ -175,6 +175,9 @@ module ChapelUtil {
   extern proc chpl_rt_preUserCodeHook();
   extern proc chpl_rt_postUserCodeHook();
 
+  extern proc allocate_string_literals_buf(s: int): c_string;
```

Comment:
So this is a wellknown function that is an extern?

When we want to add a call to a runtime function in the generated code, we add a primitive for it, and codegen it to the desired runtime function. Did you find that too heavyweight here? I found it a bit strange to have an extern proc for a function that the compiler adds the call for.

Reply:
All makes sense to me. Thanks!